haystack/components/embedders/hugging_face_tei_document_embedder.py

```diff
@@ -0,0 +1,214 @@
+        Hugging Face API token as the token parameter.
+        - **Custom TEI Endpoints**: Supports inference of embedding models deployed on custom TEI endpoints. Anyone can
+        deploy their own TEI endpoint using the TEI framework. For more details refer
+        to https://huggingface.co/inference-endpoints.
+    """
+
+    def __init__(
+        self,
+        model: str = "BAAI/bge-small-en-v1.5",
+        url: Optional[str] = None,
+        token: Optional[str] = None,
+        prefix: str = "",
+        suffix: str = "",
+        batch_size: int = 32,
+        progress_bar: bool = True,
+        metadata_fields_to_embed: Optional[List[str]] = None,
+        embedding_separator: str = "\n",
+    ):
+        """
+        Create a HuggingFaceTEIDocumentEmbedder component.
+
+        :param model: A string representing the model id on HF Hub. Default is "BAAI/bge-small-en-v1.5".
+        :param url: The URL of your self-deployed Text-Embeddings-Inference service or the URL of your paid HF Inference
+                    Endpoint.
+        :param token: The HuggingFace Hub token. This is needed if you are using a paid HF Inference Endpoint or serving
+                      a private or gated model. It can be explicitly provided or automatically read from the environment
+                      variable HF_API_TOKEN (recommended).
+        :param prefix: A string to add to the beginning of each text.
+        :param suffix: A string to add to the end of each text.
+        :param batch_size: Number of Documents to encode at once.
+        :param progress_bar: Whether to show a progress bar or not. Can be helpful to disable in production deployments
+                             to keep the logs clean.
+        :param metadata_fields_to_embed: List of meta fields that should be embedded along with the Document text.
+        :param embedding_separator: Separator used to concatenate the meta fields to the Document text.
+        """
+        transformers_import.check()
+
+        if url:
+            r = urlparse(url)
+            is_valid_url = all([r.scheme in ["http", "https"], r.netloc])
+            if not is_valid_url:
+                raise ValueError(f"Invalid TEI endpoint URL provided: {url}")
+
+        # The user does not need to provide a token if it is a local server or free public HF Inference Endpoint.
+        token = token or os.environ.get("HF_API_TOKEN")
+
+        check_valid_model(model, token)
+
+        self.model = model
+        self.url = url
+        self.token = token
+        self.client = InferenceClient(url or model, token=token)
+        self.prefix = prefix
+        self.suffix = suffix
+        self.batch_size = batch_size
+        self.progress_bar = progress_bar
+        self.metadata_fields_to_embed = metadata_fields_to_embed or []
+        self.embedding_separator = embedding_separator
+
+    def to_dict(self) -> Dict[str, Any]:
```

Comment:
We're missing `from_dict`.

Reply:
@silvanocerza I do not think the `from_dict` methods are required here, since they will follow the same behaviour as `default_from_dict`.

The implementation would just end up like this:
```python
@classmethod
def from_dict(cls, data: Dict[str, Any]) -> "HuggingFaceTEITextEmbedder":
    """
    Deserialize this component from a dictionary.
    """
    return default_from_dict(cls, data)
```